### PR TITLE
[BugFix] Fix load boolean from json

### PR DESCRIPTION
**Pull request by wyb**
## What I'm doing:
according to json type
1. boolean: true or false
2. number: !=0 is true else false
3. string: int32_t string, "true"(case insensitive) is true; "0", "false" is false, others is invalid
4. other type: invalid

Fixes https://github.com/StarRocks/StarRocksTest/issues/8095

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: be/src/formats/json/nullable_column.cpp [hunk 1]
```diff
@@ -109,7 +113,114 @@ template Status add_nullable_numeric_column<double>(Column* column, const TypeDe
 template Status add_nullable_numeric_column<float>(Column* column, const TypeDescriptor& type_desc,
                                                    const std::string& name, simdjson::ondemand::value* value);
 
-static Status add_adpative_nullable_binary_column(Column* column, const TypeDescriptor& type_desc,
+static Status add_boolean_column(Column* column, const TypeDescriptor& type_desc, const std::string& name,
```
Comment (wyb) [inline be/src/formats/json/nullable_column.cpp:116]: `add_boolean_column, add_adaptive_nullable_boolean_column, add_nullable_boolean_column` are new added, others are code refactor.

File: be/src/formats/json/nullable_column.cpp [hunk 2]
```diff
@@ -205,211 +265,178 @@ static Status add_nullable_native_json_column(Column* column, const TypeDescript
...
-
     RETURN_IF_ERROR(add_map_column(data_column.get(), type_desc, name, value));
-
     nullable_column->finish_append_one_not_default_value();
+    return Status::OK();
+}
 
+static Status add_nullable_map_column(Column* column, const TypeDescriptor& type_desc, const std::string& name,
+                                      simdjson::ondemand::value* value) {
+    DCHECK(!value->is_null());
```
Comment (luohaha) [inline be/src/formats/json/nullable_column.cpp:368]: Why this can't be null?

**Review by luohaha — DISMISSED** [review]

File: be/src/formats/json/nullable_column.cpp (hunk 2 above)
Comment (wyb) [reply be/src/formats/json/nullable_column.cpp:368]: null is processed in `add_nullable_column`
https://github.com/StarRocks/starrocks/blob/main/be/src/formats/json/nullable_column.cpp#L473

File: be/src/formats/json/nullable_column.cpp [hunk 3]
```diff
@@ -109,22 +100,111 @@ template Status add_nullable_numeric_column<double>(Column* column, const TypeDe
...
+                auto err_msg =
+                        strings::Substitute("Failed to parse string value as boolean, column=$0, value=$1", name, s);
+                return Status::InvalidArgument(err_msg);
+            }
+            bool_column->append(v);
+            return Status::OK();
+        }
 
+        default: {
+            auto err_msg = strings::Substitute("Unsupported value type. Boolean type is required. column=$0", name);
```
Comment (banmoy) [inline be/src/formats/json/nullable_column.cpp:153]: add the actual json type to err_msg?

**Review by banmoy — APPROVED** [review]

**Review by sevev — DISMISSED** [review]

File: be/src/formats/json/nullable_column.cpp (hunk 3 above)
Comment (wyb) [reply be/src/formats/json/nullable_column.cpp]: done

**Review by meegoo — APPROVED** [review]

**Review by luohaha — APPROVED** [review]

**Pull request merged**